# Define the goal
uint32 depthDemand
uint32 timeout
uint32 tolerance  
---
# Define the result
bool depthReached
---
# Define a feedback message
float32 currentDepth
